### PR TITLE
Modify windows msys runtest

### PR DESCRIPTION
**Pull request by willson-chen**
avoid of running ./fontconfig.exe or ./xpm.exe

### CODE REVIEW

**Comment by willson-chen** [comment]
related #516

**Comment by vapier** [comment]
how about we change the find back and move the logic into grep which is easier to read ?

```
for test in `find . -name \*.c | grep -vE '^./(fontconfig|gdtest|xpm)'`; do
```

**Comment by willson-chen** [comment]
> ```
> for test in `find . -name \*.c | grep -vE '^./(fontconfig|gdtest|xpm)'`; do
> ```

Aha. It works and looks more perfect.

**Comment by willson-chen** [comment]
And it seems we have missed gdhelper,

File: windows/msys/run_tests.sh [hunk 1]
```diff
@@ -9,7 +9,7 @@ CFLAGS_EXTRA=$2         # Extra C flags
 
 LOG=run_tests.log
 
-CFLAGS="-g -Igdtest/ -I. -I../src/ -D_WIN32 $CFLAGS_EXTRA"
+CFLAGS="-g -Igdtest/ -I. -I../src/ -D_WIN32 -DHAVE_SYS_STAT_H $CFLAGS_EXTRA"
```
Comment (willson-chen) [inline windows/msys/run_tests.sh]: `-DHAVE_SYS_STAT_H` is necessary for tests/gdtest/gdtest.c to include "sysy/stat.h", which defines the `struct stat` used in function _clean_dir().

**Comment by vapier** [comment]
i've manually merged this to squash the 4 redundant commits into 1.

thanks!

**Pull request closed without merging**